server/src/main/java/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -1271,6 +1280,43 @@ protected void runInContext() {
         }
     }
 
+    /**
+     * Attempts recreation of router by restarting with cleanup a VPC if any or a guest network associated in case no VPC.
+     * @param routerId - the id of the router to be recreated.
+     * @return true if successfully restart is attempted else false.
+     */
+    private boolean recreateRouter(long routerId) {
+        List<DomainRouterJoinVO> routerJoinVOs = domainRouterJoinDao.searchByIds(routerId);
+        DomainRouterJoinVO routerJoinToRestart = null;
+        User systemUser = _userDao.getUser(User.UID_SYSTEM);
+        for (DomainRouterJoinVO router : routerJoinVOs) {
+            if (router.getRemoved() == null && router.getTrafficType() == TrafficType.Guest) {
+                if (router.getVpcId() != 0) {
+                    try {
+                        s_logger.debug("Attempting restart VPC " + router.getVpcName() + " for router recreation " + router.getUuid());
+                        return vpcService.restartVpc(router.getVpcId(), true, false, systemUser);
+                    } catch (Exception e) {
+                        s_logger.error("Failed to restart VPC for router recreation " + router.getVpcName() + " ,router " + router.getUuid(), e);
+                    }
+                }
+                routerJoinToRestart = router;
+            }
```

Comment:
for all routers, if guest and present, if vpc ...
in that if it can fail and get an error logged but than it get treated as if not vpc. I don't think this is logic as intended.
Also routerJoinToRestart might get re-assigned multiple times, thus ignoring teh first one found. What is intended here?

Reply:
This is DomainRouterJoinVO for a given ID. So it's all a single router specific but for each NIC. The intention now here is we want to recreate a Router. To do that cleanly and redoing all rules we -
1. IF any network had VPC associated we just restart the VPC with cleanup and return.
2. IF no join VO had VPC we just restart any guest network with cleanup. So reassigning is fine.

Makes sense?

Reply:
Yes, it makes sense but it is ugly, and I wonder how maintainable it is.
Is it even possible that one router has a vpc and another doesn't?
So, if we found one without vpc id can we also not stop querying the rest and just restart?

Reply:
@DaanHoogland - the aim is to recreate router. My understanding was to not let any case slide and I am not sure if an entry without VPC ID would ensure it's the right one and rest entries won't have a VPC ID. Let me knw if you disagree. There are so many possible paths and it seemed like the most exhaustive approach to me. Thoughts?